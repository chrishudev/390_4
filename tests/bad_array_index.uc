void main(string[] args) {
  double f = 0.;
  args[f];
  3 + f[3];
}
