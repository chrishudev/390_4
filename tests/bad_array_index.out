Finding declarations...
Resolving types...
Resolving function calls...
Checking field and variable names...
Checking basic control flow...
Type checking...
tests/bad_array_index.uc:3:3: error (6): Cannot index non-integer double into array.
  args[f];
  ^~~~~~^
tests/bad_array_index.uc:4:7: error (6): Cannot index into non-array.
  3 + f[3];
      ^~~^
tests/bad_array_index.uc:4:7: error (6): Cannot index into non-array.
  3 + f[3];
      ^~~^
tests/bad_array_index.uc:4:3: error (6): binary + operator expects int, got double
  3 + f[3];
  ^~~~~~~^
4 errors generated in phase 6.
